### PR TITLE
Switch to GHA as a parallelization provider

### PR DESCRIPTION
**Pull request by wojteknowacki**
closes #...

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [ ] I added changesets and [read good practices](../.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [ ] customer
7. [ ] giftCard
8. [ ] homePage
9. [ ] login
10. [ ] menuNavigation
11. [ ] navigation
12. [ ] orders
13. [ ] pages
14. [ ] payments
15. [ ] permissions
16. [ ] plugins
17. [ ] productType
18. [ ] products
19. [ ] sales
20. [ ] shipping
21. [ ] translations
22. [ ] variants
23. [ ] vouchers

CONTAINERS=2

### CODE REVIEW

File: cypress.config.js [hunk 1]
```diff
@@ -26,9 +26,16 @@ module.exports = defineConfig({
...
+    setupNodeEvents(cypressOn, config) {
+      const on = require("cypress-on-fix")(cypressOn);
+      const allSpecs = findCypressSpecs();
+      const chunk = getChunk(allSpecs, k, n);
+
       config = require("./cypress/support/cypress-grep/plugin")(config);
       config = require("./cypress/plugins/index.js")(on, config);
+      config = require("cypress-split")(on, config);
+      config.specPattern = chunk;
+
```
Comment (karola312) [inline cypress.config.js:38]: It won't work. In "./cypress/support/cypress-grep/plugin" we are changing specPattern to run only specs with have tests with tags. Than you change this specPattern, so even if parallel will work, tags stop working. 
image
Can we just do that, without playing with spec pattern?

File: .github/workflows/test-env-deploy.yml [hunk 2]
```diff
@@ -251,15 +252,11 @@ jobs:
...
-          COMMIT_INFO_MESSAGE: ${{github.event.inputs.tags}} tests triggered on PR - https://github.com/${{ github.repository }}/pull/${{ github.ref_name }}
           CYPRESS_grepTags: ${{ needs.prepare-tests.outputs.tags }}
-        with:
-          parallel: true
-          group: "UI - Chrome"
-          record: true
-          tag: Critical, ${{github.event.action}}
+          SPLIT: ${{ strategy.job-total }}
+          SPLIT_INDEX: ${{ strategy.job-index }}
+
```
Comment (karola312) [inline .github/workflows/test-env-deploy.yml:259]: Add  - continue-on-error: true . When test fails next steps are skipped and results are not send do testmo

File: .github/workflows/test-env-deploy.yml [hunk 3]
```diff
@@ -241,8 +241,9 @@ jobs:
       - name: Install Dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
         run: NODE_OPTIONS=--max_old_space_size=4096 npm install
+
       - name: Cypress run critical
-        uses: cypress-io/github-action@v4
+        uses: cypress-io/github-action@v5
```
Comment (karola312) [inline .github/workflows/test-env-deploy.yml:246]: Add  - continue-on-error: true . When test fails next steps are skipped and results are not send do testmo

Comment (karola312) [inline .github/workflows/test-env-deploy.yml:246]: Add  - continue-on-error: true . When test fails next steps are skipped and results are not send do testmo

File: package.json [hunk 4]
```diff
@@ -331,7 +338,13 @@
...
     "change:add": "changeset add",
     "release:prepare-main": "./scripts/prepare-main-after-release.sh",
-    "format:code": "prettier --ignore-unknown --write src/*"
+    "format:code": "prettier --ignore-unknown --write src/*",
+    "combine:reports": "mochawesome-merge cypress/reports/mocha/*.json > cypress/reports/mochareports/report.json",
+    "generate:report": "marge cypress/reports/mochareports/report.json --reportDir cypress/reports/mochareports",
+    "html:report": "npm run combine:reports && npm run generate:report",
+    "create:artifacts:dirs": "mkdir -p cypress/reports/mochareports && mkdir -p cypress/reports/mochareports/videos",
+    "move:screenshots": "find cypress/reports/mochareports -type d -name \"*.js\" -exec mv {} cypress/reports/mochareports \\;",
+    "move:videos": "find cypress/videos -type f -name \"*.js.mp4\" -exec mv {} cypress/reports/mochareports/videos \\;"
```
Comment (andrzejewsky) [inline package.json:347]: can we combine al of the QA acions under one namespace? eg `qa:XX` -> `qa:move-videos`, `qa:create-adrifacts` etc

**Review by andrzejewsky — DISMISSED** [review]

**Review by michalina-graczyk — APPROVED** [review]

**Review by NyanKiyoshi — DISMISSED** [review]

File: .github/workflows/test-env-deploy.yml [hunk 5]
```diff
@@ -251,15 +253,27 @@ jobs:
...
           CYPRESS_USER_PASSWORD: ${{ secrets.CYPRESS_USER_PASSWORD }}
           CYPRESS_PERMISSIONS_USERS_PASSWORD: ${{ secrets.CYPRESS_PERMISSIONS_USERS_PASSWORD }}
-          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
           CYPRESS_MAILPITURL: ${{ secrets.CYPRESS_MAILPITURL }}
-          COMMIT_INFO_MESSAGE: ${{github.event.inputs.tags}} tests triggered on PR - https://github.com/${{ github.repository }}/pull/${{ github.ref_name }}
           CYPRESS_grepTags: ${{ needs.prepare-tests.outputs.tags }}
+          SPLIT: ${{ strategy.job-total }}
+          SPLIT_INDEX: ${{ strategy.job-index }}
+      - name: Create reports dir
+        if: ${{ success() || failure() }}
```
Comment (NyanKiyoshi) [inline .github/workflows/test-env-deploy.yml:261]: Docs suggest using `!cancelled()` instead which is also much shorter: https://docs.github.com/en/actions/learn-github-actions/expressions#always

```suggestion
        if: ${{ !cancelled() }}
```

File: .github/workflows/test-env-deploy.yml [hunk 6]
```diff
@@ -241,8 +241,10 @@ jobs:
       - name: Install Dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
         run: NODE_OPTIONS=--max_old_space_size=4096 npm install
+
       - name: Cypress run critical
-        uses: cypress-io/github-action@v4
+        if: ${{ success() || failure() }}
```
Comment (NyanKiyoshi) [inline .github/workflows/test-env-deploy.yml:246]:
```suggestion
        if: ${{ !cancelled() }}
```

**Review by NyanKiyoshi — APPROVED** [review]

**Review by poulch — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Pull request merged**